orbit/pkg/useraction/mdm_migration_darwin.go

```diff
@@ -563,3 +568,255 @@ func (m *swiftDialogMDMMigrator) MigrationInProgress() (bool, error) {
+	// 			log.Info().Msg("is timeout")
+	// 			return true
+	// 		}
+	// 		// Check for no such host error
+	// 		if opErr, ok := urlErr.Err.(*net.OpError); ok {
+	// 			log.Info().Msg("closing net op error")
+	// 			if dnsErr, ok := opErr.Err.(*net.DNSError); ok {
+	// 				log.Info().Msg("is dns error")
+	// 				if dnsErr.Err == "no such host" {
+	// 					log.Info().Msg("is dns no such host")
+	// 					return true
+	// 				}
+	// 			}
+	// 		}
+	// 	}
+
+	return false
+}
+
+// ShowDialogMDMMigrationOffline displays the dialog every time is called
+func (o *offlineWatcher) showSwiftDialogMDMMigrationOffline(ctx context.Context) error {
+	props := MDMMigratorProps{
+		// TODO
+	}
+	m := swiftDialogMDMMigrationOffline{
+		baseDialog: newBaseDialog(o.swiftDialogPath),
+		props:      props,
+	}
+
+	message, flags, err := m.getMessageAndFlags()
+	if err != nil {
+		return fmt.Errorf("getting offline dialog message: %w", err)
+	}
+
+	exitCodeCh, errCh := m.render(message.String(), flags...)
+
+	select {
+	case <-ctx.Done():
+		log.Info().Msg("dialog context canceled")
+		// TODO: do we care about this? anything we need to clean up?
+		return nil
+	case err := <-errCh:
+		return fmt.Errorf("showing offline dialog: %w", err)
+	case <-exitCodeCh:
+		// there's only one button, so we don't need to check the exit code
+		log.Info().Msg("closing offline dialog")
+		return nil
+	}
+}
+
+type swiftDialogMDMMigrationOffline struct {
+	*baseDialog
+	props  MDMMigratorProps
+	showCh chan struct{}
+}
+
+func (m *swiftDialogMDMMigrationOffline) render(message string, flags ...string) (chan swiftDialogExitCode, chan error) {
+	image := "/Users/Shared/Frame2.png" // TODO: update this
+	// icon := m.props.OrgInfo.OrgLogoURL
+
+	flags = append([]string{
+		// disable the built-in title so we have full control over the
+		// content
+		"--title", "none",
+		// top icon
+		"--icon", "none", // disable the built-in icon because we will render the entire content as a single image for more control over the layout
+		// "--icon", icon,
+		// "--iconsize", "80",
+		// "--centreicon",
+		// modal content
+		"--image", image,
+		// "--message", "No internet connection. Please connect to the internet to continue.",
+		// "--messagefont", "size=16",
+		"--alignment", "center",
+	}, flags...)
+
+	return m.baseDialog.render(flags...)
+}
+
+func (m *swiftDialogMDMMigrationOffline) getMessageAndFlags() (*bytes.Buffer, []string, error) {
+	tmpl := mdmMigrationOfflineTemplate
+	height := "669" // TODO: confirm this
+
+	var message bytes.Buffer
+	if err := tmpl.Execute(
+		&message,
+		m.props,
+	); err != nil {
+		return nil, nil, fmt.Errorf("executing migration template: %w", err)
+	}
+
+	flags := []string{
+		// main button
+		"--button1text", "Close",
+		"--height", height,
+	}
+
+	return &message, flags, nil
+}
+
+// TODO: Can we use a local URI for the image?
+var mdmMigrationOfflineTemplate = template.Must(template.New("mdmMigrationOfflineTemplate").Parse(`
```

Comment:
This will probably get deleted if we go with the pre-rendered image file. But I wanted to save it somewhere :)